<a href="https://colab.research.google.com/github/TosinGeorge/Data-Science-Projects/blob/main/Profitable_Mobile_English_Apps.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### **Guided Project: Profitable App Profiles for the App Store and Google Play Markets**

Introduction

The aim of this project is to recommend profitable and free mobile English apps profiles for the Google Playstore and App store. We would be exploring two datasets consisting of **10,000+** `Android` apps and **7,000+** `ios` apps. 



#### **Exploring our datasets**

In [ ]:
#importing our libraries
import pandas as pd

In [ ]:
#reading our ios and android dataset

#Google datset
android = pd.read_csv('googleplaystore (2).csv')

#ios dataset
ios = pd.read_csv('AppleStore (1).csv')

* *Determing the number of columns and rows in each dataset*

In [ ]:
#ios dataset
ios.shape

(7197, 16)

In [ ]:
#Google dataset
android.shape

(10841, 13)

**Comment**: *The google dataset has **13** columns and **10841** rows while the ios dataset has **16** 
columns and **7197** rows.*

In [ ]:
#exploring the columns names and their datatypes
android.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


In [ ]:
ios.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7197 entries, 0 to 7196
Data columns (total 16 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   id                7197 non-null   int64  
 1   track_name        7197 non-null   object 
 2   size_bytes        7197 non-null   int64  
 3   currency          7197 non-null   object 
 4   price             7197 non-null   float64
 5   rating_count_tot  7197 non-null   int64  
 6   rating_count_ver  7197 non-null   int64  
 7   user_rating       7197 non-null   float64
 8   user_rating_ver   7197 non-null   float64
 9   ver               7197 non-null   object 
 10  cont_rating       7197 non-null   object 
 11  prime_genre       7197 non-null   object 
 12  sup_devices.num   7197 non-null   int64  
 13  ipadSc_urls.num   7197 non-null   int64  
 14  lang.num          7197 non-null   int64  
 15  vpp_lic           7197 non-null   int64  
dtypes: float64(3), int64(8), object(5)
memory 

**Comment**: *The google dataset has **12** objects columns and **1** float column. There are a couple of missing values across some columns (`Rating`, `Type`, `Content Rating`, `Current Ver` and `Android Ver`). The ios datset has a mix of floats, integers and objects across across **16** columns. There are no missing values in the ios dataset. There is no need to drop the missing values in the Rating column since the column is of very little interest. We would be losing useful data if we drop the values in this column.*

In [ ]:
#exploring samples of our datasets
android.sample(2)

App  Category  Rating Reviews Size     Installs  \
2675                  CJmall  SHOPPING     3.6   18252  10M  10,000,000+   
7207  Beast of Lycan Isle CE    FAMILY     4.1    2683  20M      50,000+   

      Type Price Content Rating    Genres      Last Updated Current Ver  \
2675  Free     0       Everyone  Shopping    August 6, 2018       6.3.8   
7207  Free     0   Everyone 10+    Casual  January 23, 2014         1.0   

     Android Ver  
2675  4.1 and up  
7207  2.3 and up

**Comment**: *The columns that might be useful for our analysis are `App`, `Category`, `Reviews`, `Installs`, `Type`, `Price` and `Genre`*.



In [ ]:
ios.sample(2)

id                   track_name  size_bytes currency  price  \
1705   871694174              Toca Life: Town   453559296      USD   2.99   
4936  1100639820  HorseMoji: Equestrian Emoji    16315392      USD   1.99   

      rating_count_tot  rating_count_ver  user_rating  user_rating_ver    ver  \
1705              3174               884          4.5              4.0  1.4.1   
4936                58                32          2.0              1.5    1.3   

     cont_rating    prime_genre  sup_devices.num  ipadSc_urls.num  lang.num  \
1705          4+      Education               40                5         1   
4936          4+  Entertainment               37                5         1   

      vpp_lic  
1705        1  
4936        1

**Comment:** *The columns that might be useful for our analysis are `track_name`, `currency`, `price`, `rating_count_tot`, `rating_count_ver` and `prime_genre`*.

#### **Deleting Wrong data and duplicated values**

*The Google Play Store dataset has a dedicated discussion section that describes an error for a certain row. I printed the affected row including the row before and after so I could easily make comparison.*



In [ ]:
android[10471:10474]

App         Category  Rating  \
10471                         Xposed Wi-Fi-Pwd  PERSONALIZATION     3.5   
10472  Life Made WI-Fi Touchscreen Photo Frame              1.9    19.0   
10473                  osmino Wi-Fi: free WiFi            TOOLS     4.2   

      Reviews    Size     Installs  Type     Price Content Rating  \
10471    1042    404k     100,000+  Free         0       Everyone   
10472    3.0M  1,000+         Free     0  Everyone            NaN   
10473  134203    4.1M  10,000,000+  Free         0       Everyone   

                  Genres    Last Updated Current Ver   Android Ver  
10471    Personalization  August 5, 2014       3.0.0  4.0.3 and up  
10472  February 11, 2018          1.0.19  4.0 and up           NaN  
10473              Tools  August 7, 2018     6.06.14    4.4 and up

In [ ]:
#dropping the row
android=android.drop(10472)

#### **Removing Duplicate Entries**

*I also checked for duplicate entries across each datset.*

In [ ]:
android.duplicated('App').sum()

1181

In [ ]:
ios.duplicated('track_name').sum()

2

**Comment:** *There are **1181** duplicate apps name in the google dataset and **2** duplicate apps name in the ios dataset* 

In [ ]:
#Exploring some of the duplicate app names in the Google Dataset
android.App.value_counts()


ROBLOX                                                9
CBS Sports App - Scores, News, Stats & Watch Live     8
ESPN                                                  7
Duolingo: Learn Languages Free                        7
Candy Crush Saga                                      7
                                                     ..
Meet U - Get Friends for Snapchat, Kik & Instagram    1
U-Report                                              1
U of I Community Credit Union                         1
Waiting For U Launcher Theme                          1
iHoroscope - 2018 Daily Horoscope & Astrology         1
Name: App, Length: 9659, dtype: int64

**Comment:** *`ROBLOX`, `CBS Sports App - Scores, News, Stats & Watch Live`,`ESPN`, `Duolingo: Learn Languages Free` and `Candy Crush Saga` are some of the duplicated app names in the google dataset*

In [ ]:
#Exploring one of the duplicated app name. Roblox exists 9 times in the Google dataset
android.groupby('App').get_group('ROBLOX')

App Category  Rating  Reviews Size      Installs  Type Price  \
1653  ROBLOX     GAME     4.5  4447388  67M  100,000,000+  Free     0   
1701  ROBLOX     GAME     4.5  4447346  67M  100,000,000+  Free     0   
1748  ROBLOX     GAME     4.5  4448791  67M  100,000,000+  Free     0   
1841  ROBLOX     GAME     4.5  4449882  67M  100,000,000+  Free     0   
1870  ROBLOX     GAME     4.5  4449910  67M  100,000,000+  Free     0   
2016  ROBLOX   FAMILY     4.5  4449910  67M  100,000,000+  Free     0   
2088  ROBLOX   FAMILY     4.5  4450855  67M  100,000,000+  Free     0   
2206  ROBLOX   FAMILY     4.5  4450890  67M  100,000,000+  Free     0   
4527  ROBLOX   FAMILY     4.5  4443407  67M  100,000,000+  Free     0   

     Content Rating                        Genres   Last Updated  \
1653   Everyone 10+  Adventure;Action & Adventure  July 31, 2018   
1701   Everyone 10+  Adventure;Action & Adventure  July 31, 2018   
1748   Everyone 10+  Adventure;Action & Adventure  July 31, 2018   
1841   Everyone 10+  Adventure;Action & Adventure  July 31, 2018   
1870   Everyone 10+  Adventure;Action & Adventure  July 31, 2018   
2016   Everyone 10+  Adventure;Action & Adventure  July 31, 2018   
2088   Everyone 10+  Adventure;Action & Adventure  July 31, 2018   
2206   Everyone 10+  Adventure;Action & Adventure  July 31, 2018   
4527   Everyone 10+  Adventure;Action & Adventure  July 31, 2018   

       Current Ver Android Ver  
1653  2.347.225742  4.1 and up  
1701  2.347.225742  4.1 and up  
1748  2.347.225742  4.1 and up  
1841  2.347.225742  4.1 and up  
1870  2.347.225742  4.1 and up  
2016  2.347.225742  4.1 and up  
2088  2.347.225742  4.1 and up  
2206  2.347.225742  4.1 and up  
4527  2.347.225742  4.1 and up

In [ ]:
#Exploring some of the duplicate app names in the ios Dataset
ios.track_name.value_counts()

VR Roller Coaster                                   2
Mannequin Challenge                                 2
Facebook                                            1
iFleeing                                            1
Fear for Sale: The House on Black River (Full)      1
                                                   ..
PDF Converter - Convert Documents, Photos to PDF    1
Motion Math: Hungry Fish                            1
MindNode – Delightful Mind Mapping                  1
Skylanders SuperChargers                            1
みんなのお弁当 by クックパッド ~お弁当をレシピ付きで記録・共有~                 1
Name: track_name, Length: 7195, dtype: int64

**Comment:** *The duplicated app names across ios datsets are `VR Roller Coaster` and `Mannequin Challenge`.*

In [ ]:
ios.groupby('track_name').get_group('VR Roller Coaster')

id         track_name  size_bytes currency  price  \
4442   952877179  VR Roller Coaster   169523200      USD    0.0   
4831  1089824278  VR Roller Coaster   240964608      USD    0.0   

      rating_count_tot  rating_count_ver  user_rating  user_rating_ver    ver  \
4442               107               102          3.5              3.5  2.0.0   
4831                67                44          3.5              4.0   0.81   

     cont_rating prime_genre  sup_devices.num  ipadSc_urls.num  lang.num  \
4442          4+       Games               37                5         1   
4831          4+       Games               38                0         1   

      vpp_lic  
4442        1  
4831        1

* Dropping duplicates

In [ ]:
#I sorted the values in the Reviews columns by descending order, identified duplicates app names and dropped them. 
android_clean = android.sort_values('Reviews', ascending=False).drop_duplicates('App').sort_index()
android_clean

App             Category  \
0         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
2      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                  Sketch - Draw & Paint       ART_AND_DESIGN   
4                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
5                             Paper flowers instructions       ART_AND_DESIGN   
...                                                  ...                  ...   
10836                                   Sya9a Maroc - FR               FAMILY   
10837                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                             Parkinson Exercices FR              MEDICAL   
10839                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating Reviews                Size     Installs  Type Price  \
0         4.1     159                 19M      10,000+  Free     0   
2         4.7   87510                8.7M   5,000,000+  Free     0   
3         4.5  215644                 25M  50,000,000+  Free     0   
4         4.3     967                2.8M     100,000+  Free     0   
5         4.4     167                5.6M      50,000+  Free     0   
...       ...     ...                 ...          ...   ...   ...   
10836     4.5      38                 53M       5,000+  Free     0   
10837     5.0       4                3.6M         100+  Free     0   
10838     NaN       3                9.5M       1,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   
10840     4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating                   Genres      Last Updated  \
0           Everyone             Art & Design   January 7, 2018   
2           Everyone             Art & Design    August 1, 2018   
3               Teen             Art & Design      June 8, 2018   
4           Everyone  Art & Design;Creativity     June 20, 2018   
5           Everyone             Art & Design    March 26, 2017   
...              ...                      ...               ...   
10836       Everyone                Education     July 25, 2017   
10837       Everyone                Education      July 6, 2018   
10838       Everyone                  Medical  January 20, 2017   
10839     Mature 17+        Books & Reference  January 19, 2015   
10840       Everyone                Lifestyle     July 25, 2018   

              Current Ver         Android Ver  
0                   1.0.0        4.0.3 and up  
2                   1.2.4        4.0.3 and up  
3      Varies with device          4.2 and up  
4                     1.1          4.4 and up  
5                     1.0          2.3 and up  
...                   ...                 ...  
10836                1.48          4.1 and up  
10837                 1.0          4.1 and up  
10838                 1.0          2.2 and up  
10839  Varies with device  Varies with device  
10840  Varies with device  Varies with device  

[9659 rows x 13 columns]

In [ ]:
ios_clean = ios.sort_values('rating_count_tot', ascending=False).drop_duplicates('track_name').sort_index()
ios_clean

id                                    track_name  size_bytes  \
0      284882215                                      Facebook   389879808   
1      389801252                                     Instagram   113954816   
2      529479190                                Clash of Clans   116476928   
3      420009108                                    Temple Run    65921024   
4      284035177                       Pandora - Music & Radio   130242560   
...          ...                                           ...         ...   
7192  1170406182  Shark Boom - Challenge Friends with your Pet   245415936   
7193  1069830936                           【謎解き】ヤミすぎ彼女からのメッセージ    16808960   
7194  1070052833                                    Go!Go!Cat!    91468800   
7195  1081295232   Suppin Detective: Expose their true visage!    83026944   
7196   977965019           みんなのお弁当 by クックパッド ~お弁当をレシピ付きで記録・共有~    51174400   

     currency  price  rating_count_tot  rating_count_ver  user_rating  \
0         USD    0.0           2974676               212          3.5   
1         USD    0.0           2161558              1289          4.5   
2         USD    0.0           2130805               579          4.5   
3         USD    0.0           1724546              3842          4.5   
4         USD    0.0           1126879              3594          4.0   
...       ...    ...               ...               ...          ...   
7192      USD    0.0                 0                 0          0.0   
7193      USD    0.0                 0                 0          0.0   
7194      USD    0.0                 0                 0          0.0   
7195      USD    0.0                 0                 0          0.0   
7196      USD    0.0                 0                 0          0.0   

      user_rating_ver      ver cont_rating        prime_genre  \
0                 3.5     95.0          4+  Social Networking   
1                 4.0    10.23         12+      Photo & Video   
2                 4.5  9.24.12          9+              Games   
3                 4.0    1.6.2          9+              Games   
4                 4.5    8.4.1         12+              Music   
...               ...      ...         ...                ...   
7192              0.0    1.0.9          4+              Games   
7193              0.0      1.2          9+               Book   
7194              0.0    1.1.2         12+              Games   
7195              0.0    1.0.3         12+      Entertainment   
7196              0.0    1.4.0          4+       Food & Drink   

      sup_devices.num  ipadSc_urls.num  lang.num  vpp_lic  
0                  37                1        29        1  
1                  37                0        29        1  
2                  38                5        18        1  
3                  40                5         1        1  
4                  37                4         1        1  
...               ...              ...       ...      ...  
7192               38                5         1        1  
7193               38                0         1        1  
7194               37                2         2        1  
7195               40                0         1        1  
7196               37                0         1        1  

[7195 rows x 16 columns]

**Comment**: *After dropping the wrong data row and missing values, we are left with **9659** rows in the Google dataset and **7195** rows in the ios dataset.* 

In [ ]:
android_clean.duplicated('App').sum()

0

In [ ]:
ios_clean.duplicated('track_name').sum()

0

* For the next phase of cleaning, I decided to convert each of the datasets to lists.

In [ ]:
#Conversion to lists
android_list = android_clean.values.tolist()
ios_list = ios_clean.values.tolist()

### **Remove Non-English Apps**

This project is focused on the English-speaking audience and as such I would have to exclude non-English apps.

Letters and special characters in the English Alphabet have a ASCII number between 1 and 127. Based off this knowledge, we can try to exclude non-english apps. Some English apps have non-ASCIII characters and dropping them from our data will impact data loss. To minimize the impact of data loss, we'll only remove an app if its name has more than three non-ASCII characters.

   


In [ ]:
def is_english(string):
    non_ascii = 0
    
    for character in string:
        if ord(character) > 127:
            non_ascii += 1
    
    if non_ascii > 3:
        return False
    else:
        return True

print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

True
True


* Filtering for English apps names across both datasets.

In [ ]:
android_english = []
ios_english = []

for app in android_list:
    name = app[0]
    if is_english(name):
        android_english.append(app)
print ('English Android Apps:',len(android_english))

for app in ios_list:
    name = app[1]
    if is_english(name):
        ios_english.append(app)
print ('IOS Android Apps:',len(ios_english))


English Android Apps: 9614
IOS Android Apps: 6181



> **Comment:** We have **9614** and **6181** English apps in our Google store and App store datasets






* I converted our datasets back to a panda dataframe.

In [ ]:
android_data = pd.DataFrame(android_english,columns =['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver'])
ios_data = pd.DataFrame(ios_english, columns = ['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic'])

### **Free English Apps**

In [ ]:
#identifying free android apps
android_free = android_data.loc[android_data['Price'] == '0']
len(android_free)

8862

In [ ]:
#identifying free ios apps
ios_free = ios_data.loc[ios_data['price'] == 0.0]
len(ios_free)

3220

> **Comment:** We have **8862** and **3220** free English apps in our Google store and App store datasets






#### **Data Exploration & Analysis**

The objective of this project is to identify the free most profitable English app profiles across the Google and Apple app stores. To determine the profiles, I will be exploring popular apps by Genres and no of downloads(`Installs` and `user rating count`).

* **Most Popular Genres (Google Playstore)**


In [ ]:
#Exploring the "Category" columns with a frequency table.

freq = pd.crosstab(index = android_free['Category'], columns='count')
freq= freq/len(android_free)*100
freq= freq.sort_values(by=['count'], ascending=False)
freq

col_0                    count
Category                      
FAMILY               18.946062
GAME                  9.693072
TOOLS                 8.451817
BUSINESS              4.592643
LIFESTYLE             3.904311
PRODUCTIVITY          3.893026
FINANCE               3.701196
MEDICAL               3.520650
SPORTS                3.396524
PERSONALIZATION       3.317536
COMMUNICATION         3.238547
HEALTH_AND_FITNESS    3.080569
PHOTOGRAPHY           2.945159
NEWS_AND_MAGAZINES    2.798465
SOCIAL                2.663056
TRAVEL_AND_LOCAL      2.335816
SHOPPING              2.245543
BOOKS_AND_REFERENCE   2.143986
DATING                1.861882
VIDEO_PLAYERS         1.794177
MAPS_AND_NAVIGATION   1.399233
FOOD_AND_DRINK        1.241255
EDUCATION             1.173550
ENTERTAINMENT         0.947867
LIBRARIES_AND_DEMO    0.936583
AUTO_AND_VEHICLES     0.925299
HOUSE_AND_HOME        0.823742
WEATHER               0.801174
EVENTS                0.710900
PARENTING             0.654480
ART_AND_DESIGN        0.643196
COMICS                0.620627
BEAUTY                0.598059

**Comment:** *18.9% of the apps on Google Playstore are dominated by family-oriented apps, followed by Games(9.69%), Tools(8.45%), Business(4.59%) and Lifestyle(3.9%). I will attempt to explore the top dominant apps below based on the number of installations.*

The `Installs` column estimates the number of installs per app and as such it's difficult to get the exact number of downloads. For now, we assume the values of the `Installs` column as the exact number of downloads.

Since the `Installs` column is an object type. I converted the values to floats so that we can perform the necessary mathematical operations.

In [ ]:
import warnings
warnings.filterwarnings('ignore')

android_free['Installs']= android_free.Installs.str.replace(',', '', regex = True)
android_free['Installs']=android_free.Installs.str.replace('+', '', regex = True)
android_free['Installs'] = android_free['Installs'].astype(float)

In [ ]:
#Exploring the FAMILY category

and_family = android_free[android_free["Category"]=='FAMILY']
and_family = and_family.groupby('App')['Installs'].sum().sort_values( ascending=False)
and_family[:20]

App
Google Play Games                           1.000000e+09
Bitmoji – Your Personal Emoji               1.000000e+08
Talking Tom Cat 2                           1.000000e+08
Farm Heroes Saga                            1.000000e+08
Tom Loves Angela                            1.000000e+08
Bubble Witch 2 Saga                         1.000000e+08
Where's My Water? Free                      1.000000e+08
Hay Day                                     1.000000e+08
Duolingo: Learn Languages Free              1.000000e+08
Minion Rush: Despicable Me Official Game    1.000000e+08
Talking Tom Cat                             1.000000e+08
ROBLOX                                      1.000000e+08
Talking Tom & Ben News                      1.000000e+08
Talking Ginger                              1.000000e+08
SpongeBob Diner Dash                        5.000000e+07
Boom Beach                                  5.000000e+07
Vegas Crime Simulator                       5.000000e+07
LEGO® Juniors Create & Crui

**Comment:** The top 20 family apps by downloads is basically a variety of games for children.


In [ ]:
#Exploring the GAME category

and_game = android_free[android_free["Category"]=='GAME']
and_game = and_game.groupby('App')['Installs'].sum().sort_values( ascending=False)
and_game[:20]

App
Subway Surfers                          1.000000e+09
Pou                                     5.000000e+08
Temple Run 2                            5.000000e+08
Candy Crush Saga                        5.000000e+08
My Talking Tom                          5.000000e+08
Extreme Car Driving Simulator           1.000000e+08
Piano Tiles 2™                          1.000000e+08
Helix Jump                              1.000000e+08
Ant Smasher by Best Cool & Fun Games    1.000000e+08
Angry Birds Star Wars                   1.000000e+08
Angry Birds Rio                         1.000000e+08
Angry Birds Classic                     1.000000e+08
Banana Kong                             1.000000e+08
Angry Birds 2                           1.000000e+08
Hill Climb Racing                       1.000000e+08
Super Mario Run                         1.000000e+08
Geometry Dash Lite                      1.000000e+08
PAC-MAN                                 1.000000e+08
Agar.io                                 1.

In [ ]:
#Exploring the TOOLS category

and_tools = android_free[android_free["Category"]=='TOOLS']
and_tools = and_tools.groupby('App')['Installs'].sum().sort_values( ascending=False)
and_tools[:10]

App
Google                                                1.000000e+09
Security Master - Antivirus, VPN, AppLock, Booster    5.000000e+08
Gboard - the Google Keyboard                          5.000000e+08
SHAREit - Transfer & Share                            5.000000e+08
Google Translate                                      5.000000e+08
Clean Master- Space Cleaner & Antivirus               5.000000e+08
GO Keyboard - Cute Emojis, Themes and GIFs            1.000000e+08
Samsung Calculator                                    1.000000e+08
AVG AntiVirus 2018 for Android Security               1.000000e+08
Samsung Smart Switch Mobile                           1.000000e+08
Name: Installs, dtype: float64

In [ ]:
#Exploring the BUSINESS category

and_biz = android_free[android_free["Category"]=='BUSINESS']
and_biz = and_biz.groupby('App')['Installs'].sum().sort_values( ascending=False)
and_biz[:10]

App
File Commander - File Manager/Explorer                100000000.0
OfficeSuite : Free Office + PDF Editor                100000000.0
Facebook Pages Manager                                 50000000.0
Vault-Hide SMS,Pics & Videos,App Lock,Cloud backup     50000000.0
Indeed Job Search                                      50000000.0
Docs To Go™ Free Office Suite                          50000000.0
Secure Folder                                          50000000.0
Skype for Business for Android                         10000000.0
Fast Scanner : Free PDF Scan                           10000000.0
MyASUS - Service Center                                10000000.0
Name: Installs, dtype: float64

In [ ]:
#Exploring the LIFESTYLE category

and_ls = android_free[android_free["Category"]=='LIFESTYLE']
and_ls = and_ls.groupby('App')['Installs'].sum().sort_values( ascending=False)
and_ls[:10]

App
Tinder                                                100000000.0
Samsung+                                               50000000.0
Diary with lock                                        10000000.0
Zara                                                   10000000.0
Daily Horoscope                                        10000000.0
Sleep as Android: Sleep cycle tracker, smart alarm     10000000.0
Rate Guide Bill Letter                                 10000000.0
OK cashbag [point of pleasure]                         10000000.0
Talking Babsy Baby: Baby Games                         10000000.0
Muslim Pro - Prayer Times, Azan, Quran & Qibla         10000000.0
Name: Installs, dtype: float64

**Comment:** *The most popular genres for the Google Playstore are Family, Games, Tools, Business and Lifestyle. I would recommend creating app profiles from these categories. However, developing an app because it belongs to a popular genre does not mean that it would generate a lot of downloads.*

I would attempt to explore the different genres by the number of downloads. 

* **Top Downloads by Genre (Google Playstore)**





In [ ]:
pop_data = android_free.groupby('Category')['Installs'].sum()
pop_data = pop_data/android_free['Category'].value_counts()
pop_data.sort_values(ascending=False)

COMMUNICATION          3.845612e+07
VIDEO_PLAYERS          2.472787e+07
SOCIAL                 2.325365e+07
PHOTOGRAPHY            1.780563e+07
PRODUCTIVITY           1.678733e+07
GAME                   1.556097e+07
TRAVEL_AND_LOCAL       1.398408e+07
ENTERTAINMENT          1.171976e+07
TOOLS                  1.068230e+07
NEWS_AND_MAGAZINES     9.549178e+06
BOOKS_AND_REFERENCE    8.767812e+06
SHOPPING               7.036877e+06
PERSONALIZATION        5.201483e+06
WEATHER                5.074486e+06
HEALTH_AND_FITNESS     4.188822e+06
MAPS_AND_NAVIGATION    4.056942e+06
FAMILY                 3.695054e+06
SPORTS                 3.638640e+06
ART_AND_DESIGN         1.986335e+06
FOOD_AND_DRINK         1.924898e+06
EDUCATION              1.820673e+06
BUSINESS               1.712290e+06
LIFESTYLE              1.437816e+06
FINANCE                1.387692e+06
HOUSE_AND_HOME         1.331541e+06
DATING                 8.540288e+05
COMICS                 8.176573e+05
AUTO_AND_VEHICLES      6.473

**Comments:** *As expected the Communications category received a lot of downloads, followed by Video Players, Social, Photography and Productivity*

In [ ]:
# Exploring the "COMMUNICATION" category
and_comms = android_free[android_free["Category"]=='COMMUNICATION']
and_comms = and_comms.groupby('App')['Installs'].sum().sort_values( ascending=False)
and_comms[:20]

App
Google Chrome: Fast & Secure                         1.000000e+09
Messenger – Text and Video Chat for Free             1.000000e+09
Hangouts                                             1.000000e+09
Gmail                                                1.000000e+09
WhatsApp Messenger                                   1.000000e+09
Skype - free IM & video calls                        1.000000e+09
Google Duo - High Quality Video Calls                5.000000e+08
Viber Messenger                                      5.000000e+08
imo free video calls and chat                        5.000000e+08
LINE: Free Calls & Messages                          5.000000e+08
UC Browser - Fast Download Private & Secure          5.000000e+08
WeChat                                               1.000000e+08
Telegram                                             1.000000e+08
Who                                                  1.000000e+08
Opera Mini - fast web browser                        1.000000e+08
Opera 

In [ ]:
# Exploring the "VIDEO PLAYERS" category

and_vids = android_free[android_free["Category"]=='VIDEO_PLAYERS']
and_vids = and_vids.groupby('App')['Installs'].sum().sort_values( ascending=False)
and_vids[:20]

App
Google Play Movies & TV                               1.000000e+09
YouTube                                               1.000000e+09
MX Player                                             5.000000e+08
VivaVideo - Video Editor & Photo Movie                1.000000e+08
VideoShow-Video Editor, Video Maker, Beauty Camera    1.000000e+08
VLC for Android                                       1.000000e+08
Motorola Gallery                                      1.000000e+08
Motorola FM Radio                                     1.000000e+08
Dubsmash                                              1.000000e+08
MiniMovie - Free Video and Slideshow Editor           5.000000e+07
LIKE – Magic Video Maker & Community                  5.000000e+07
DU Recorder – Screen Recorder, Video Editor, Live     5.000000e+07
Ringdroid                                             5.000000e+07
KineMaster – Pro Video Editor                         5.000000e+07
VMate                                                 5.00

In [ ]:
# Exploring the "SOCIAL" category

and_social = android_free[android_free["Category"]=='SOCIAL']
and_social = and_social.groupby('App')['Installs'].sum().sort_values( ascending=False)
and_social[:20]

App
Instagram                                 1.000000e+09
Google+                                   1.000000e+09
Facebook                                  1.000000e+09
Snapchat                                  5.000000e+08
Facebook Lite                             5.000000e+08
BIGO LIVE - Live Stream                   1.000000e+08
LinkedIn                                  1.000000e+08
Tumblr                                    1.000000e+08
Tik Tok - including musical.ly            1.000000e+08
Pinterest                                 1.000000e+08
VK                                        1.000000e+08
Tango - Live Video Broadcast              1.000000e+08
Badoo - Free Chat & Dating App            1.000000e+08
MeetMe: Chat & Meet New People            5.000000e+07
Zello PTT Walkie Talkie                   5.000000e+07
POF Free Dating App                       5.000000e+07
ooVoo Video Calls, Messaging & Stories    5.000000e+07
SKOUT - Meet, Chat, Go Live               5.000000e+07
Messen

In [ ]:
# Exploring the "PHOTOGRAGHY" category

and_photo = android_free[android_free["Category"]=='PHOTOGRAPHY']
and_photo = and_photo.groupby('App')['Installs'].sum().sort_values( ascending=False)
and_photo[:20]

App
Google Photos                                           1.000000e+09
Photo Editor Pro                                        1.000000e+08
Retrica                                                 1.000000e+08
PhotoGrid: Video & Pic Collage Maker, Photo Editor      1.000000e+08
Candy Camera - selfie, beauty camera, photo editor      1.000000e+08
Camera360: Selfie Photo Editor with Funny Sticker       1.000000e+08
Cymera Camera- Photo Editor, Filter,Collage,Layout      1.000000e+08
BeautyPlus - Easy Photo Editor & Selfie Camera          1.000000e+08
Photo Collage Editor                                    1.000000e+08
S Photo Editor - Collage Maker , Photo Collage          1.000000e+08
Sweet Selfie - selfie camera, beauty cam, photo edit    1.000000e+08
B612 - Beauty & Filter Camera                           1.000000e+08
PicsArt Photo Studio: Collage Maker & Pic Editor        1.000000e+08
AR effect                                               1.000000e+08
YouCam Makeup - Magic Selfie M

In [ ]:
# Exploring the "PRODUCTIVITY" category

and_prod = android_free[android_free["Category"]=='PRODUCTIVITY']
and_prod = and_prod.groupby('App')['Installs'].sum().sort_values( ascending=False)
and_prod[:20]

App
Google Drive                                         1.000000e+09
Google Calendar                                      5.000000e+08
Cloud Print                                          5.000000e+08
Microsoft Word                                       5.000000e+08
Dropbox                                              5.000000e+08
Microsoft PowerPoint                                 1.000000e+08
WPS Office - Word, Docs, PDF, Note, Slide & Sheet    1.000000e+08
Google Slides                                        1.000000e+08
Evernote – Organizer, Planner for Notes & Memos      1.000000e+08
SwiftKey Keyboard                                    1.000000e+08
Adobe Acrobat Reader                                 1.000000e+08
Samsung Notes                                        1.000000e+08
Microsoft Outlook                                    1.000000e+08
Microsoft OneNote                                    1.000000e+08
CamScanner - Phone PDF Creator                       1.000000e+08
Micros

In [ ]:
# Exploring the "TRAVEL" category

and_trav = android_free[android_free["Category"]=='TRAVEL_AND_LOCAL']
and_trav = and_trav.groupby('App')['Installs'].sum().sort_values( ascending=False)
and_trav[:20]

App
Maps - Navigate & Explore                             1.000000e+09
Google Street View                                    1.000000e+09
TripAdvisor Hotels Flights Restaurants Attractions    1.000000e+08
Google Earth                                          1.000000e+08
Booking.com Travel Deals                              1.000000e+08
VZ Navigator                                          5.000000e+07
MAPS.ME – Offline Map and Travel Navigation           5.000000e+07
trivago: Hotels & Travel                              5.000000e+07
2GIS: directory & navigator                           5.000000e+07
Goibibo - Flight Hotel Bus Car IRCTC Booking App      1.000000e+07
GasBuddy: Find Cheap Gas                              1.000000e+07
Despegar.com Hotels and Flights                       1.000000e+07
Skyscanner                                            1.000000e+07
KAYAK Flights, Hotels & Cars                          1.000000e+07
KakaoMap - Map / Navigation                           1.00

In [ ]:
# Exploring the "PHOTOGRAGHY" category

and_photo = android_free[android_free["Category"]=='NEWS_AND_MAGAZINES']
and_photo = and_photo.groupby('App')['Installs'].sum().sort_values( ascending=False)
and_photo[:20]

App
Google News                                            1.000000e+09
Flipboard: News For Our Time                           5.000000e+08
Twitter                                                5.000000e+08
Dailyhunt (Newshunt) - Latest News, Viral Videos       5.000000e+07
Opera News - Trending news and videos                  1.000000e+07
News by The Times of India Newspaper - Latest News     1.000000e+07
NewsDog - Latest News, Breaking News, Local News       1.000000e+07
Updates for Samsung - Android Update Versions          1.000000e+07
BBC News                                               1.000000e+07
Topbuzz: Breaking News, Videos & Funny GIFs            1.000000e+07
Newsroom: News Worth Sharing                           1.000000e+07
News Republic                                          1.000000e+07
CNN Breaking US & World News                           1.000000e+07
BaBe - Read News                                       1.000000e+07
Pulse Nabd - World News, Urgent             

In [ ]:
and_br = android_free[android_free["Category"]=='BOOKS_AND_REFERENCE']
and_br = and_br.groupby('App')['Installs'].sum().sort_values( ascending=False)
and_br[:30]

App
Google Play Books                                     1.000000e+09
Wattpad 📖 Free Books                                  1.000000e+08
Audiobooks from Audible                               1.000000e+08
Amazon Kindle                                         1.000000e+08
Bible                                                 1.000000e+08
Al'Quran Bahasa Indonesia                             1.000000e+07
Al Quran Indonesia                                    1.000000e+07
English Hindi Dictionary                              1.000000e+07
English Dictionary - Offline                          1.000000e+07
HTC Help                                              1.000000e+07
JW Library                                            1.000000e+07
Aldiko Book Reader                                    1.000000e+07
Dictionary.com: Find Definitions for English Words    1.000000e+07
Al-Quran (Free)                                       1.000000e+07
Moon+ Reader                                          1.00

**Comment:** *Most of the apps recording high downloads are owned by big players such as Google, Microsoft, Amazon,and Youtube. It will be difficult to compete with these players, particularly in the Communications, Social, Games, Photography, Productivity and, even, travel.Asides the big players, there are a host of other well known apps recording millions of downloads.*

* **Most Popular Genres (App Store)**


In [ ]:
#exploring the "prime_genre" columns

freq = pd.crosstab(index=ios_free['prime_genre'], columns='count')
freq= freq/len(ios_free)*100
freq= freq.sort_values(by=['count'], ascending=False)
freq

col_0                  count
prime_genre                 
Games              58.136646
Entertainment       7.888199
Photo & Video       4.968944
Education           3.664596
Social Networking   3.291925
Shopping            2.608696
Utilities           2.515528
Sports              2.142857
Music               2.049689
Health & Fitness    2.018634
Productivity        1.739130
Lifestyle           1.583851
News                1.335404
Travel              1.242236
Finance             1.118012
Weather             0.869565
Food & Drink        0.807453
Reference           0.559006
Business            0.527950
Book                0.434783
Navigation          0.186335
Medical             0.186335
Catalogs            0.124224

**Comment:** *More than half (58.1%) of the apps on the App Store are Games, followed by Entertainment(7.88%), Photo & Video(4.96%), Education(3.66%) and Social Networking(3.29%). I will explore the top dominant apps below based on the count of ratings.*


In [ ]:
#exploring the "Games" columns

ios_games = ios_free[ios_free["prime_genre"]=='Games']
ios_games = ios_games.groupby('track_name')['rating_count_tot'].sum().sort_values( ascending=False)
ios_games[:20]

track_name
Clash of Clans                             2130805
Temple Run                                 1724546
Candy Crush Saga                            961794
Angry Birds                                 824451
Subway Surfers                              706110
Solitaire                                   679055
CSR Racing                                  677247
Crossy Road - Endless Arcade Hopper         669079
Injustice: Gods Among Us                    612532
Hay Day                                     567344
PAC-MAN                                     508808
DragonVale                                  503230
Head Soccer                                 481564
Despicable Me: Minion Rush                  464312
The Sims™ FreePlay                          446880
Sonic Dash                                  418033
8 Ball Pool™                                416736
Tiny Tower - Free City Building             414803
Jetpack Joyride                             405647
Bike Race - Top Moto

In [ ]:
#exploring the "Entertainment" columns

ios_ent = ios_free[ios_free["prime_genre"]=='Entertainment']
ios_ent = ios_ent.groupby('track_name')['rating_count_tot'].sum().sort_values( ascending=False)
ios_ent[:20]

track_name
Netflix                                            308844
Fandango Movies - Times + Tickets                  291787
Colorfy: Coloring Book for Adults                  247809
IMDb Movies & TV - Trailers and Showtimes          183425
TRUTH or DARE!!! - FREE                            171055
Mad Libs                                           117889
Twitch                                             109549
Action Movie FX                                    101222
Voice Changer Plus                                  98777
iFunny :)                                           98344
The CW                                              97368
The Moron Test                                      88613
DIRECTV                                             81006
ABC – Watch Live TV & Stream Full Episodes          78890
Xbox                                                72187
Redbox                                              60236
Talking Tom Cat 2 for iPad                          56399
Hul

In [ ]:
#exploring the "Photo & Video" columns

ios_pv = ios_free[ios_free["prime_genre"]=='Photo & Video']
ios_pv = ios_pv.groupby('track_name')['rating_count_tot'].sum().sort_values( ascending=False)
ios_pv[:20]

track_name
Instagram                                             2161558
Snapchat                                               323905
YouTube - Watch Videos, Music, and Live Streams        278166
Pic Collage - Picture Editor & Photo Collage Maker     123433
Funimate video editor: add cool effects to videos      123268
musical.ly - your video social network                 105429
Photo Collage Maker & Photo Editor - Live Collage       93781
Vine Camera                                             90355
Google Photos - unlimited photo and video storage       88742
Flipagram                                               79905
Mixgram - Picture Collage Maker - Pic Photo Editor      54282
Shutterfly: Prints, Photo Books, Cards Made Easy        51427
Pic Jointer – Photo Collage, Camera Effects Editor      51330
Color Pop Effects - Photo Editor & Picture Editing      45320
Photo Grid - photo collage maker & photo editor         40531
iSwap Faces LITE                                        397

In [ ]:
#exploring the "Education" columns

ios_education = ios_free[ios_free["prime_genre"]=='Education']
ios_education = ios_education.groupby('track_name')['rating_count_tot'].sum().sort_values( ascending=False)
ios_education[:20]

track_name
Duolingo - Learn Spanish, French and more            162701
Guess My Age  Math Magic                            123190
Lumosity - Brain Training                             96534
Elevate - Brain Training and Games                    58092
Fit Brains Trainer                                    46363
ClassDojo                                             35440
Memrise: learn languages                              20383
Peak - Brain Training                                 20322
Canvas by Instructure                                 19981
ABCmouse.com - Early Learning Academy                 18749
Quizlet: Study Flashcards, Languages & Vocabulary     16683
Photomath - Camera Calculator                         16523
iTunes U                                              15801
Blackboard Mobile Learn™                              13567
Star Chart                                            13482
Remind: Fast, Efficient School Messaging               9796
PBS KIDS Video               

* **Top Downloads by Genre (App Store)**




In [ ]:
#average downloads by Genre in the ios dataset

pop_data = ios_free.groupby('prime_genre')['rating_count_tot'].sum()
pop_data = pop_data/ios_free['prime_genre'].value_counts()
pop_data.sort_values(ascending=False)

Navigation           86090.333333
Reference            74942.111111
Social Networking    71548.349057
Music                57326.530303
Weather              52279.892857
Book                 39758.500000
Food & Drink         33333.923077
Finance              31467.944444
Photo & Video        28441.543750
Travel               28243.800000
Shopping             26919.690476
Health & Fitness     23298.015385
Sports               23008.898551
Games                22812.924679
News                 21248.023256
Productivity         21028.410714
Utilities            18684.456790
Lifestyle            16485.764706
Entertainment        14029.830709
Business              7491.117647
Education             7003.983051
Catalogs              4004.000000
Medical                612.000000
dtype: float64

In [ ]:
#Exploring the "Navigation" columns

ios_nav = ios_free[ios_free["prime_genre"]=='Navigation']
ios_nav = ios_nav.groupby('track_name')['rating_count_tot'].sum().sort_values( ascending=False)
ios_nav[:10]

track_name
Waze - GPS Navigation, Maps & Real-time Traffic     345046
Google Maps - Navigation & Transit                  154911
Geocaching®                                          12811
CoPilot GPS – Car Navigation & Offline Maps           3582
ImmobilienScout24: Real Estate Search in Germany       187
Railway Route Search                                     5
Name: rating_count_tot, dtype: int64

In [ ]:
#exploring the "Reference" columns

ios_ref = ios_free[ios_free["prime_genre"]=='Reference']
ios_ref = ios_ref.groupby('track_name')['rating_count_tot'].sum().sort_values( ascending=False)
ios_ref[:15]

track_name
Bible                                                                                                 985920
Dictionary.com Dictionary & Thesaurus                                                                 200047
Dictionary.com Dictionary & Thesaurus for iPad                                                         54175
Google Translate                                                                                       26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran                                                     18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition                                 17588
Merriam-Webster Dictionary                                                                             16849
Night Sky                                                                                              12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE)                          8535
LUCKY BL

In [ ]:
#exploring the "Music" columns

ios_mus = ios_free[ios_free["prime_genre"]=='Music']
ios_mus = ios_mus.groupby('track_name')['rating_count_tot'].sum().sort_values( ascending=False)
ios_mus[:15]

track_name
Pandora - Music & Radio                              1126879
Spotify Music                                         878563
Shazam - Discover music, artists, videos & lyrics     402925
iHeartRadio – Free Music & Radio Stations             293228
SoundCloud - Music & Audio                            135744
Magic Piano by Smule                                  131695
Smule Sing!                                           119316
TuneIn Radio - MLB NBA Audiobooks Podcasts Music      110420
Amazon Music                                          106235
SoundHound Song Search & Music Player                  82602
Sonos Controller                                       48905
Bandsintown Concerts                                   30845
Karaoke - Sing Karaoke, Unlimited Songs!               28606
My Mixtapez Music                                      26286
Sing Karaoke Songs Unlimited with StarMaker            26227
Name: rating_count_tot, dtype: int64

In [ ]:
#exploring the "Weather" columns

ios_wea = ios_free[ios_free["prime_genre"]=='Weather']
ios_wea = ios_wea.groupby('track_name')['rating_count_tot'].sum().sort_values( ascending=False)
ios_wea[:15]

track_name
The Weather Channel: Forecast, Radar & Alerts                                            495626
The Weather Channel App for iPad – best local forecast, radar map, and storm tracking    208648
WeatherBug - Local Weather, Radar, Maps, Alerts                                          188583
MyRadar NOAA Weather Radar Forecast                                                      150158
AccuWeather - Weather for Life                                                           144214
Yahoo Weather                                                                            112603
Weather Underground: Custom Forecast & Local Radar                                        49192
NOAA Weather Radar - Weather Forecast & HD Radar                                          45696
Weather Live Free - Weather Forecast & Alerts                                             35702
Storm Radar                                                                               22792
QuakeFeed Earthquake Map, Ale

In [ ]:
#exploring the "Social Networking" columns

ios_soc = ios_free[ios_free["prime_genre"]=='Social Networking']
ios_soc = ios_soc.groupby('track_name')['rating_count_tot'].sum().sort_values( ascending=False)
ios_soc[:15]

track_name
Facebook                                              2974676
Pinterest                                             1061624
Skype for iPhone                                       373519
Messenger                                              351466
Tumblr                                                 334293
WhatsApp Messenger                                     287589
Kik                                                    260965
ooVoo – Free Video Call, Text and Voice                177501
TextNow - Unlimited Text + Calls                       164963
Viber Messenger – Text & Call                          164249
Followers - Social Analytics For Instagram             112778
MeetMe - Chat and Meet New People                       97072
We Heart It - Fashion, wallpapers, quotes, tattoos      90414
InsTrack for Instagram - Analytics Plus More            85535
Tango - Free Video Call, Voice and Chat                 75412
Name: rating_count_tot, dtype: int64

In [ ]:
#exploring the "Travel" columns

ios_trav = ios_free[ios_free["prime_genre"]=='Travel']
ios_trav = ios_trav.groupby('track_name')['rating_count_tot'].sum().sort_values( ascending=False)
ios_trav[:15]

track_name
Google Earth                                        446185
Yelp - Nearby Restaurants, Shopping & Services      223885
GasBuddy                                            145549
TripAdvisor Hotels Flights Restaurants               56194
Uber                                                 49466
Lyft                                                 46922
HotelTonight - Great Deals on Last Minute Hotels     32341
Hotels & Vacation Rentals by Booking.com             31261
Southwest Airlines                                   30552
Airbnb                                               22302
Expedia Hotels, Flights & Vacation Package Deals     10278
Fly Delta                                             8094
Hopper - Predict, Watch & Book Flights                6944
United Airlines                                       5748
Skiplagged — Actually Cheap Flights & Hotels          1851
Name: rating_count_tot, dtype: int64

### **Conclusion:**

I will recommend apps profiles from the **Travel** and **Reference** genres for the App store and **Book & References** genre for the Google Playstore. Since Games is one of the most popular genres across both stores. I would recommend creating games apps around popular books like the Bible, Quran and Dictionaries. 